compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/selection/SelectionManager.uikit.kt

```diff
@@ -16,11 +16,44 @@
 
 package androidx.compose.foundation.text.selection
 
+import androidx.compose.foundation.PlatformMagnifierFactory
+import androidx.compose.foundation.isPlatformMagnifierSupported
+import androidx.compose.foundation.magnifier
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
 import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.IntSize
 
 internal actual fun isCopyKeyEvent(keyEvent: KeyEvent): Boolean =
     false //TODO implement copy key event for iPad
 
-internal actual fun Modifier.selectionMagnifier(manager: SelectionManager): Modifier =
-    this //TODO support magnifier for uikit
+internal actual fun Modifier.selectionMagnifier(manager: SelectionManager): Modifier {
+    if (!isPlatformMagnifierSupported()) {
+        return this
+    }
+
+    return composed {
+        val density = LocalDensity.current
+        var magnifierSize by remember { mutableStateOf(IntSize.Zero) }
+        val color = LocalTextSelectionColors.current
+
+        magnifier(
+            sourceCenter = {
+                // Don't animate position as it is automatically animated by the framework
+                calculateSelectionMagnifierCenterAndroid(manager, magnifierSize)
```

Comment:
Should `calculateSelectionMagnifierCenterIOS`  also be called be here?

Reply:
There 2 incompatible `calculateSelectionMagnifierCenterAndroid` implementations for `TextFieldSelectionManager` and `SelectionManager`. As for `SelectionContainer` iOS magnifier behavior is almost the same as on Android so i just left android implementation here.